src/extension.ts

```diff
@@ -8,10 +8,52 @@ import * as vscode from 'vscode';
 // this method is called when your extension is activated
 // your extension is activated the very first time the command is executed
 export async function activate(context: vscode.ExtensionContext) {
+    console.log("activated!")
+    // Add status bar button
+    let statusBarJsim = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 100);
+    let statusBarBsim = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 100);
+	statusBarJsim.command = "compstruct-vscode.jsimTerminal";
+    statusBarBsim.command = "compstruct-vscode.bsimTerminal";
+    statusBarJsim.text = "$(play) Run JSim";
+    statusBarBsim.text = "$(play) Run BSim";
+    statusBarJsim.show();
+    statusBarBsim.show();
+	context.subscriptions.push(statusBarJsim);
+    context.subscriptions.push(statusBarBsim);
 
+    context.subscriptions.push(vscode.commands.registerCommand('compstruct-vscode.jsimTerminal', () => {
+        runJar("jsim")
+	}));
+
+    context.subscriptions.push(vscode.commands.registerCommand('compstruct-vscode.bsimTerminal', () => {
+        runJar("bsim")
+	}));
 }
 
 // this method is called when your extension is deactivated
 export function deactivate() {
 
 }
+
+function runJar(jarname: string) {
+    // Find terminal if exists
+    let terminal;
+    for (const term of vscode.window.terminals) {
+        const name = term.name
+        if (name === jarname) {
+            terminal = term;
+        }
+    }
+    if (!terminal) {
+        terminal = vscode.window.createTerminal(jarname);
+    }
+    const filepath = vscode.window.activeTextEditor?.document.uri.fsPath;
```

Comment:
Hmm... Does this mean that this function will only run the `jsim.jar` and `bsim.jar` files if they are in the root directory of the current workspace? In that case, since we want this extension to be general enough and since different people might organize their files differently, perhaps one suggestion that I have would be to recursively search for the jar files in all the directories and subdirectories of the current workspace, and then if the jar files with the appropriate names are found, it will execute them? Otherwise, it should also show a helpful pop-up message indicating that it failed to find the jar files in the current workspace so that the current user would be in the know of the extension's status.